packages/fx-core/src/plugins/resource/bot/aadRegistration.ts

```diff
@@ -72,18 +72,28 @@ export class AADRegistration {
 
   public static async registerAADAppAndGetSecretByAppStudio(
     appStudioToken: string,
-    displayName: string
+    displayName: string,
+    objectId?: string,
+    msAppId?: string
   ): Promise<BotAuthCredential> {
     const result = new BotAuthCredential();
 
     const appConfig: IAADDefinition = {
       displayName: displayName,
     };
 
-    const app = await AppStudio.createAADAppV2(appStudioToken, appConfig);
-    result.clientId = app.appId;
-    result.objectId = app.id;
+    if (!objectId && !msAppId) {
+      const app = await AppStudio.createAADAppV2(appStudioToken, appConfig);
+      result.clientId = app.appId;
+      result.objectId = app.id;
+    } else {
+      CheckThrowSomethingMissing("objectId", objectId);
+      CheckThrowSomethingMissing("msAppId", msAppId);
+      result.objectId = objectId;
+      result.clientId = msAppId;
+    }
 
+    // if has objectId, just create password for this AAD
```

Comment:
The comment needs to be adapted. When it comes to this line, result.objectId is definitely non-null, so remove `if has objectId` in the comment or just remove the line of comment totally.